pandas/_libs/src/ujson/python/objToJSON.c

```diff
@@ -233,12 +251,18 @@ static PyObject *get_values(PyObject *obj) {
 
 static PyObject *get_sub_attr(PyObject *obj, char *attr, char *subAttr) {
     PyObject *tmp = PyObject_GetAttrString(obj, attr);
+    if (tmp == NULL) {
+        return NULL;
+    }
     PyObject *ret;
 
     if (tmp == 0) {
         return 0;
     }
     ret = PyObject_GetAttrString(tmp, subAttr);
+    if (ret == NULL) {
+        return ret;
+    }
     Py_DECREF(tmp);
```

Comment:
I guess this is a case where it could make sense to have say `Py_DECREF` right before the error checking; whether or not things worked you are down with the lifecycle of `tmp`